cmake/download/download_and_check.cmake

```diff
@@ -22,14 +22,19 @@ function (DownloadAndCheck from to fatal result sha256)
           Download(${from} ${to} ${fatal} ${result} output ${sha256})
           list(GET output 0 status_code)
         else()
-          message(STATUS "${WGET_EXECUTABLE} --no-cache --no-check-certificate 
-            --retry-connrefused --waitretry=1 --read-timeout=20 --timeout=15 --tries=5 ${from}")
-          execute_process(COMMAND ${WGET_EXECUTABLE} "--no-cache" "--no-check-certificate" 
-            "--retry-connrefused" "--waitretry=1" "--read-timeout=20" "--timeout=15" "--tries=5" 
-            "${from}" "-O" "${to}"
-            TIMEOUT 2000
-            RESULT_VARIABLE status_code)
-          file(SHA256 ${to} CHECKSUM)
+          foreach(index RANGE 15)
+            message(STATUS "${WGET_EXECUTABLE} --no-cache --no-check-certificate 
+              --retry-connrefused --waitretry=1 --read-timeout=20 --timeout=15 --tries=5 ${from}")
+            execute_process(COMMAND ${WGET_EXECUTABLE} "--no-cache" "--no-check-certificate" 
+              "--retry-connrefused" "--waitretry=1" "--read-timeout=20" "--timeout=15" "--tries=5" 
+              "${from}" "-O" "${to}"
+              TIMEOUT 2000
```

Comment:
Am I right that we want to run downloading process 15 times (if previous runs are not successfull) and  each run have 5 attempts to download?

Reply:
@ilyachur , seems like just 15 attempts.

BTW, why 15? Seems like too much for me. Maybe 3 times?

Reply:
@ilyachur wget option "--tries 5" doesn't help in case of proxy error, downloading fails with "Unable to establish SSL connection" error without any retries.
Okay, I'll reduce the number of iterations.